sparrow-py/src/execution.rs

```diff
@@ -0,0 +1,31 @@
+use arrow::pyarrow::ToPyArrow;
+use pyo3::prelude::*;
+use sparrow_session::Execution as RustExecution;
+
+use crate::error::{Error, ErrorContext};
+
+/// Kaskada execution object.
+#[pyclass]
+pub(crate) struct Execution(Option<RustExecution>);
+
+impl Execution {
+    pub(crate) fn new(execution: RustExecution) -> Self {
+        Self(Some(execution))
+    }
+}
+
+#[pymethods]
+impl Execution {
+    fn collect_pyarrow(&mut self, py: Python<'_>) -> Result<Vec<PyObject>, Error> {
```

Comment:
Could the be confused with `Expr.collect()`?

Reply:
I don't think so? This is part of `sparrow_py._ffi.Execution`, while that is a method on `sparrow-py.expr.Expr`, and has a different name (`collect` vs. `collect_pyarrow`).

Reply:
It's not a big deal, but in one case we're describing an operation and in another, we're controlling execution. I could imagine the execution class moving around or becoming more "public". Not a big deal - I was just wondering if there's a similar word we could use.

Reply:
Ah. I think it's pretty common for the "collect the results of this thing"... I think we're probably OK -- one being called `collect` and the other `collect_pyarrow` or `collect_dataframes` etc. In terms of the user, it's about whether they've already run or not:

```
collect().run()
vs.
run().collect_dataframes()
```